### ISSUE TITLE
Training stops after 1st epoch on contrastive finetuning

### ISSUE THREAD

**Issue by DanielMitiku** [nomic-ai/contrastors#44]
The training stops after the first epoch. No error is shown but no training happens after the first epoch. I have attached the data_args here.

data_args:
  input_shards: "configs/data/finetune_triplets.yaml"
  workers: 8 
  batch_size: 256
  seed: 42
  shuffle: false
  download: true
  streaming: true
  weighted_sampling: false
  verbose: true

**Comment by zanussbaum** [comment]
hey @DanielMitiku you should be able to set the number of epochs > 1 here: https://github.com/nomic-ai/contrastors/blob/main/src/contrastors/configs/train/contrastive_finetune.yaml#L2

let me know if this doesn't work. i also saw diminishing returns training for longer but YMMV

**Comment by DanielMitiku** [comment]
Yeah, I already did that. But it won't train after 1st epoch.

**Comment by zanussbaum** [comment]
Can you describe a little more what you see after the first epoch?

**Comment by DanielMitiku** [comment]
During the first epoch, training will continue normally for all the steps but after the first epoch ends, with in seconds these are printed out and training finishes.

[Epoch 1:   0%|          | 0/6612 [00:00<?, ?it/s]
Epoch 1:   0%|          | 0/6612 [00:00<?, ?it/s]

Epoch 2:   0%|          | 0/6612 [00:00<?, ?it/s]
Epoch 2:   0%|          | 0/6612 [00:00<?, ?it/s]

Epoch 3:   0%|          | 0/6612 [00:00<?, ?it/s]
Epoch 3:   0%|          | 0/6612 [00:00<?, ?it/s]

Epoch 4:   0%|          | 0/6612 [00:00<?, ?it/s]
Epoch 4:   0%|          | 0/6612 [00:00<?, ?it/s]

**Comment by zanussbaum** [comment]
Thanks! Taking a look

**Comment by zanussbaum** [comment]
Thanks for finding this bug! Fixed in [9089049](https://github.com/nomic-ai/contrastors/commit/9089049b391e79a80ec37531096de84029b7aa5b)
